index.js

```diff
@@ -112,6 +141,56 @@ class Conf {
 		}
 	}
 
+	_migrate(migrations, versionToMigrate) {
+		const MIGRATION_KEY = '__internal__.version';
+
+		const previousMigratedVersion = this.get(MIGRATION_KEY, '0.0.0');
+
+		if (semver.eq(previousMigratedVersion, versionToMigrate)) {
+			return;
+		}
+
+		const newerVersions = Object.keys(migrations)
+			.filter(candidateVersion => this._shouldPerformMigration(candidateVersion, previousMigratedVersion, versionToMigrate))
+			.sort(semver.compare);
+
+		newerVersions.forEach(version => {
+			try {
+				const migration = migrations[version];
+				migration(this);
+
+				this.set(MIGRATION_KEY, version);
+			} catch (error) {
+				console.error('Something went wrong during the migration!');
+				console.error('Changes applied to the store until this point will be persisted.');
+				throw error;
+			}
+		});
+
+		this.set(MIGRATION_KEY, versionToMigrate);
```

Comment:
This operation still executes 1 time unnecessary in the following scenario:

```
previousMigratedVersion = 1.0.1
versionToMigrate = 1.0.1
```

It will run `this.set(MIGRATION_KEY, 1.0.1);` inside the `forEach` loop and again here. To solve this, simply move the `semver.eq` check here and update `previousMigratedVersion` inside the loop

```js
               // inside the loop..
               this.set(MIGRATION_KEY, version);
               previousMigratedVersion = version;

        // outside the loop, at the bottom..
	if (!semver.eq(previousMigratedVersion, versionToMigrate)) {
		this.set(MIGRATION_KEY, versionToMigrate);
	}
```

Reply:
You're right since we're filtering the versions, the early exit is a kind of pointless. However, one could argue that it does prevent the filter from iterating the migrations object.

Updated it, thanks 👍